set http_proxy environment variables on controller pod

Kishorb opened the pull request:
Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

fawadkhaliq commented on stable/appmesh-controller/templates/deployment.yaml:
```diff
@@ -71,6 +71,13 @@ spec:
         - --aws-region={{ .Values.region }}
         {{- end }}
         - --sidecar-log-level={{ .Values.sidecar.logLevel }}
+        {{- if .Values.proxyEnv }}
```
Since `proxyEnv` can set any `env`, I'd suggest renaming `proxyEnv` to `env` for accuracy and extensibility

Kishorb replied:
Thank you for your comment and updated as per your recommendation. Could you please check now. Thank you.

Kishorb commented:
> @Kishorb thanks for your contribution! can you please also update the README. Also left a minor comment

@fawadkhaliq - I renamed  proxyEnv to env  as per your recommendation and also added the variable to README. Could you please check now.

Kishorb commented:
Hello @fawadkhaliq - all conflicts resolved now, could you please review my PR now. Thank you.

jujugrrr commented:
@fawadkhaliq Hi, any chance to get this merged? thanks a lot

fawadkhaliq commented:
+1 /lgtm
